File: src/main/vrml/net/coderextreme/x3d_code/www_web3d_org/x3d/content/examples/Basic/CAD/QuadSetExample.js
let browser = X3D.getBrowser();
let X3D0 = {};
X3D0.profile = "CADInterchange";
X3D0.version = "3.1";
let WorldInfo2 = browser.currentScene.createNode("WorldInfo");
WorldInfo2.title = "QuadSetExample.x3d";
browser.currentScene.children = new MFNode();

browser.currentScene.children[0] = WorldInfo2;

let NavigationInfo3 = browser.currentScene.createNode("NavigationInfo");
NavigationInfo3.type = new MFString(new java.lang.String["EXAMINE"]);
browser.currentScene.children[1] = NavigationInfo3;

let Viewpoint4 = browser.currentScene.createNode("Viewpoint");
Viewpoint4.DEF = "Side";
Viewpoint4.description = "QuadSet side view";
browser.currentScene.children[2] = Viewpoint4;

let Viewpoint5 = browser.currentScene.createNode("Viewpoint");
Viewpoint5.DEF = "Front";
Viewpoint5.description = "QuadSet front view";
Viewpoint5.orientation = new SFRotation(new float[0,1,0,1.57]);
Viewpoint5.position = new SFVec3f(new float[10,0,0]);
browser.currentScene.children[3] = Viewpoint5;

let Viewpoint6 = browser.currentScene.createNode("Viewpoint");
Viewpoint6.DEF = "Top";
Viewpoint6.description = "QuadSet top view";
Viewpoint6.orientation = new SFRotation(new float[1,0,0,-1.57]);
Viewpoint6.position = new SFVec3f(new float[0,10,0]);
browser.currentScene.children[4] = Viewpoint6;

let Transform7 = browser.currentScene.createNode("Transform");
Transform7.scale = new SFVec3f(new float[2,2,2]);
let Shape8 = browser.currentScene.createNode("Shape");
let Appearance9 = browser.currentScene.createNode("Appearance");
let Material10 = browser.currentScene.createNode("Material");
Material10.diffuseColor = new SFColor(new float[0.321569,0.592157,1]);
Appearance9.material = Material10;

Shape8.appearance = Appearance9;

let QuadSet11 = browser.currentScene.createNode("QuadSet");
QuadSet11.solid = False;
let Coordinate12 = browser.currentScene.createNode("Coordinate");
Coordinate12.point = new MFVec3f(new float[-1.5,0,0,-0.5,0,0,-0.5,1,-1,-1.5,1,-1,0.5,0,0,1.5,0,0,1.5,1,-1,0.5,1,-1,-0.5,2,-1,0.5,2,-1]);
QuadSet11.coord = Coordinate12;

Shape8.geometry = QuadSet11;

Transform7.children = new MFNode();

Transform7.children[0] = Shape8;

browser.currentScene.children[5] = Transform7;

File: src/main/python/net/x3djsonld/data/c/x3d-code/www.web3d.org/x3d/content/examples/Basic/CAD/QuadSetExample.py
###############################################
#
# Now available: developmental python x3d.py package on PyPi for import.
#   This approach greatly simplifies Python X3D deployment and use.
#   https://pypi.org/project/x3d
#
# Installation:
#       pip install x3d
# or
#       python -m pip install x3d
#
# Developer options for loading x3d package:
#
#    from x3d import *  # preferred approach, terser source that avoids x3d.* class prefixes
#
# or
#    import x3d         # traditional way to subclass x3d package, all classes require x3d.* prefix
#                       # but python source is very verbose, for example x3d.Material x3d.Shape etc.
#                       # X3dToPython.xslt stylesheet insertPackagePrefix=true supports this option.

from x3d import *

###############################################

newModel=X3D(profile='CADInterchange',version='3.1',
  head=head(
    children=[
    component(level=2,name='CADGeometry'),
    meta(content='QuadSetExample.x3d',name='title'),
    meta(content='Simple QuadSet example scene.',name='description'),
    meta(content='Leonard Daly, Don Brutzman',name='creator'),
    meta(content='19 April 2013',name='created'),
    meta(content='20 October 2019',name='modified'),
    meta(content='https://www.web3d.org/x3d/content/examples/Basic/CAD/QuadSetExample.x3d',name='identifier'),
    meta(content='https://www.web3d.org/files/specifications/19775-1/V3.3/Part01/components/CADGeometry.html#QuadSet',name='reference'),
    meta(content='https://www.web3d.org/files/specifications/19775-1/V3.3/Images/quadSet.png',name='reference'),
    meta(content='X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit',name='generator'),
    meta(content='../license.html',name='license')]),
  Scene=Scene(
    children=[
    WorldInfo(title='QuadSetExample.x3d'),
    NavigationInfo(type=["EXAMINE"]),
    Viewpoint(DEF='Side',description='QuadSet side view'),
    Viewpoint(DEF='Front',description='QuadSet front view',orientation=(0,1,0,1.57),position=(10,0,0)),
    Viewpoint(DEF='Top',description='QuadSet top view',orientation=(1,0,0,-1.57),position=(0,10,0)),
    Transform(scale=(2,2,2),
      children=[
      Shape(
        appearance=Appearance(
          material=Material(diffuseColor=(0.321569,0.592157,1))),
        geometry=QuadSet(ccw=True,colorPerVertex=True,normalPerVertex=True,solid=False,
          coord=Coordinate(point=[(-1.5,0,0),(-0.5,0,0),(-0.5,1,-1),(-1.5,1,-1),(0.5,0,0),(1.5,0,0),(1.5,1,-1),(0.5,1,-1),(-0.5,2,-1),(0.5,2,-1)])))])])
) # X3D model complete

###############################################
# Self-test diagnostics
###############################################

if        metaDiagnostics(newModel): # built-in utility method in X3D class
    print(metaDiagnostics(newModel))
print('check  newModel.XML() serialization...')
newModelXML = newModel.XML() # test export method XML() for exceptions
# print(newModelXML) # debug

print ("python x3d.py load successful for QuadSetExample.py")
